File: src/LinksNews/Services/Data/Models/NewsSource.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace LinksNews.Services.Data.Models
{
    [Table("vwNewsSource")]
    public class NewsSource
    {
        public long? Id { get; set; }

        public long? NewsProviderId { get; set; }
        public string NewsProvider { get; set; }

        public string NewsSourceId { get; set; }
        public string NewsSourceDescription { get; set; }
        public string NewsSourceUrl { get; set; }

        public string NewsSourceSmallLogoUrl { get; set; }
        public string NewsSourceMediumLogoUrl { get; set; }
        public string NewsSourceLargeLogoUrl { get; set; }

        public long? ContentCategoryId { get; set; }
        public string ContentCategory { get; set; }

        public long? CountryId { get; set; }
        public string CountryCode { get; set; }
        public string CountryName { get; set; }

        public long? LanguageId { get; set; }
        public string LanguageCode { get; set; }
        public string LanguageName { get; set; }
    }
}

File: src/LinksNews/angular2App/app/core/domain/news-source.ts
import * as _ from 'lodash';
import { Link } from "./page/link";

export class NewsSource {
    id: number;

    newsProviderId: number;
    newsProvider: string;

    newsSourceId: string;
    newsSourceDescription: string;
    newsSourceUrl: string;

    newsSourceSmallLogoUrl: string;
    newsSourceMediumLogoUrl: string;
    newsSourceLargeLogoUrl: string;
 
    contentCategoryId: number;
    contentCategory: string;
    contentCategoryTranslated: string;
 
    countryId: number;
    countryCode: string;
    countryName: string;
    countryNameTranslated: string;
 
    languageId: number;
    languageCode: string;
    languageName: string;
    languageNameTranslated: string;
        
    selected: boolean = false;

    links: Array<Link>;


    clone(): NewsSource {
        let result: NewsSource = <NewsSource>_.assign(new NewsSource(), this);
        return result;
    }

    static from (newsSource: NewsSource): NewsSource {
        
        let result: NewsSource = new NewsSource();
        _.assign(result, newsSource);
        return result;
    }
}